### PR TITLE
Implementation of defer, first cut.

### PR DESCRIPTION
**Pull request by thanm**
Includes functional versions of the Backend interfaces for
function_defer_statement() and exception_handler_statement(),
along with initial unit tests for these. Needs additional testing.

### CODE REVIEW

File: llvm-gofrontend/go-llvm-typemanager.cpp [hunk 1]
```diff
@@ -1199,6 +1199,27 @@ Btype *TypeManager::circularTypeAddrConversion(Btype *typ) {
...
+  llvm::SmallVector<llvm::Type *, 2> elems(2);
+  elems[0] = llvmPtrType();
+  elems[1] = llvmInt32Type();
+  return llvm::StructType::get(context_, elems);
+}
+
+llvm::FunctionType *TypeManager::personalityFunctionType()
+{
+  const bool isVarargs = false;
+  llvm::SmallVector<llvm::Type *, 4> elems(4);
```
Comment (ianlancetaylor) [inline llvm-gofrontend/go-llvm-typemanager.cpp]: That's interesting, the GCC personality function has five arguments.  I wonder why they are different?

    _Unwind_Reason_Code
    PERSONALITY_FUNCTION (int version,
    		      _Unwind_Action actions,
    		      _Unwind_Exception_Class exception_class,
    		      struct _Unwind_Exception *ue_header,
    		      struct _Unwind_Context *context);

File: llvm-gofrontend/go-llvm.cpp [hunk 2]
```diff
@@ -3179,12 +3291,217 @@ llvm::BasicBlock *GenBlocks::genSwitch(Bstatement *swst,
...
+
+  // Pop the pad block stack
+  padBlockStack_.pop_back();
+
+  // If the body block ended without a return, then insert a jump
+  // to the continue / or finally clause.
+  if (curblock)
+    llvm::BranchInst::Create(contbb, curblock);
+
+  // Emit landing pad inst in pad block, followed by branch to catch bb
```
Comment (ianlancetaylor) [inline llvm-gofrontend/go-llvm.cpp]: Missing period at end of comment sentence.

**Review by ianlancetaylor — APPROVED** [review]
LGTM

File: llvm-gofrontend/go-llvm-typemanager.cpp (hunk 1 above)
Comment (thanm) [reply llvm-gofrontend/go-llvm-typemanager.cpp]: Good catch, thanks.

File: llvm-gofrontend/go-llvm.cpp (hunk 2 above)
Comment (thanm) [reply llvm-gofrontend/go-llvm.cpp]: Thanks, will fix that up.

**Pull request merged**